avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/AvroParserImplDecoder.java

```diff
@@ -138,6 +138,13 @@ public void skipString() throws IOException {
     @Override
     public ByteBuffer readBytes(ByteBuffer old) throws IOException {
         consumeToken(JsonToken.VALUE_EMBEDDED_OBJECT);
+        byte[] value = _parser.getBinaryValue();
```

Comment:
Bugfix: Caller expecting `old` to be populated and not actually using the return value needs to be supported.